ganttproject/src/main/java/net/sourceforge/ganttproject/chart/ChartModelBase.java

```diff
@@ -221,6 +241,33 @@ public void languageChanged(Event event) {
         timeFormatters.setLocaleApi(localeApi);
       }
     });
+
+    // clear formatter caches
+    myWeekNumberOption.addChangeValueListener(evt -> {timeFormatters.setLocaleApi(localeApi);});
+    myTaskManager.addTaskListener(new TaskListenerAdapter() {
+      @Override
+      public void taskScheduleChanged(@NotNull TaskScheduleEvent e) {
+        resetLocaleApi();
+      }
+      @Override
+      public void taskAdded(@NotNull TaskHierarchyEvent e) {
+        resetLocaleApi();
+      }
+      @Override
+      public void taskMoved(@NotNull TaskHierarchyEvent e) {
+        resetLocaleApi();
+      }
+      @Override
+      public void taskRemoved(@NotNull TaskHierarchyEvent e) {
+        resetLocaleApi();
+      }
+      private void resetLocaleApi() {
+        if (myWeekNumberOption.isChecked()) {
+          timeFormatters.setLocaleApi(localeApi);
```

Comment:
This way we will reset cache excessively. I suggest replacing LocaleApi::getWeekNumber(Date) with `getWeekNumberProperty(): SimpleObjectProperty<Function<Date, Integer>>`

Change the function when the option changes, and subscribe to the property changes in the week formatter.

Reply:
Observable property class has been created